### ISSUE TITLE
`map_matching_delta` may be used inappropriately

### ISSUE THREAD

**Issue by ezheidtmann** [Project-OSRM/osrm-backend#4808]
I was reviewing the map matching plugin and found a comparison that doesn't make sense to me.

The `max_distance_delta` is set to be the maximum distance that could possibly have been traveled during the current time step:

```
# (paraphrasing)
const auto max_distance_delta = step_time * facade.GetMapMatchingMaxSpeed();
```

And `d_t` is the absolute difference between the great-circle distance and the network distance. 

```
const auto d_t = std::abs(network_distance - haversine_distance);
```

Then we [compare the two](https://github.com/Project-OSRM/osrm-backend/blob/master/src/engine/routing_algorithms/map_matching.cpp#L236) to throw out very low probability transitions:

```
                    // very low probability transition -> prune
                    if (d_t >= max_distance_delta)
                    {
                        continue;
                    }
```

I think this is wrong; if the actual speed was close to the maximum, we might still consider transitions that have a speed 2x the facade's maximum speed.

Instead, maybe compare `max_distance_delta` to `network_distance`? If the network distance implies travel over the maximum speed, then it should be thrown out.

**Comment by TheMarex** [comment]
Good catch, this should rather be the `network_distance`. Your fix should improve the performance quite a bit! Do you want to prepare a PR for this?
